src/itensor.jl

```diff
@@ -1982,22 +1983,21 @@ Scale the ITensor A by x in-place. May also be written `rmul!`.
 A .*= x
 ```
 """
-scale!(T::ITensor, α::Number) = (T .*= α)
+scale!(T::ITensor, α::Number) = itensor(data(T) .*= α, inds(T))
 
-rmul!(T::ITensor, α::Number) = (T .*= α)
+rmul!(T::ITensor, α::Number) = itensor(data(T) .*= α, inds(T))
 
-lmul!(T::ITensor, α::Number) = (T .= α .* T)
+lmul!(T::ITensor, α::Number) = itensor(data(T) .= α .* data(T), inds(T))
 
 """
     mul!(A::ITensor, x::Number, B::ITensor)
 
 Scalar multiplication of ITensor B with x, and store the result in A.
 Like `A .= x .* B`.
 """
-mul!(R::ITensor, α::Number, T::ITensor) = (R .= α .* T)
-
-mul!(R::ITensor, T::ITensor, α::Number) = (R .= T .* α)
+mul!(R::ITensor, α::Number, T::ITensor) = itensor(data(R) .= α * data(T), inds(T))
 
+mul!(R::ITensor, T::ITensor, α::Number) = itensor(data(R) .= data(T) * α, inds(T))
```

Comment:
Again, these will all be wrong for sparse tensors so we need to fix it in `NDTensors`.